packages/components/src/pages/Quickstart.vue

```diff
@@ -0,0 +1,775 @@
+import StepRightIcon from '@/assets/quickstart/arrow-right.svg';
+import SuccessIcon from '@/assets/quickstart/success.svg';
+
+export const Steps = {
+  INSTALL_AGENT: 1,
+  CONFIGURE_APPMAP: 2,
+  RECORD_APPMAPS: 3,
+};
+
+export default {
+  name: 'Quickstart',
+
+  components: {
+    HelpIcon,
+    StepLeftIcon,
+    StepRightIcon,
+    SuccessIcon,
+  },
+
+  props: {
+    language: {
+      type: String,
+      default: null,
+    },
+    testFrameworks: {
+      type: Array,
+      default: () => [],
+    },
+    installSnippets: {
+      type: Object,
+      default: () => {},
+    },
+    appmapYmlSnippet: {
+      type: String,
+      default: '',
+    },
+    initialStep: {
+      type: Number,
+      default: 1,
+    },
+    stepsState: {
+      type: Array,
+      default: () => [],
+    },
+    appmapsProgress: {
+      type: Number,
+      default: 0,
+    },
+    onAction: {
+      type: Function,
+    },
+    onStep: {
+      type: Function,
+    },
+    error: {
+      type: String,
+    },
+  },
+
+  data() {
+    return {
+      selectedLanguage: this.language,
+      selectedTestFramework: this.testFrameworks.length
+        ? this.testFrameworks[0]
+        : null,
+      currentStep: this.initialStep,
+      isActionRunning: false,
+      showProjectSelector: false,
+      projects: [],
+      selectedProject: null,
+    };
+  },
+
+  watch: {
+    currentStep: {
+      handler() {
+        if (typeof this.onStep === 'function') {
+          this.onStep(this.currentStep);
+        }
+      },
+    },
+  },
+
+  computed: {
+    stepsCount() {
+      return Object.keys(Steps).length;
+    },
+    step1Completed() {
+      return this.stepsState[0] === 'complete';
+    },
+    step2Completed() {
+      return this.stepsState[1] === 'complete';
+    },
+    step3Completed() {
+      return this.stepsState[2] === 'complete';
+    },
+    languages() {
+      const languages = {
+        ruby: 'Ruby',
+        java: 'Java',
+        python: 'Python',
+      };
+
+      if (Object.keys(languages).includes(this.language)) {
+        languages[this.language] += ' (detected)';
+      }
+
+      return languages;
+    },
+    appmapsProgressText() {
+      return `${this.appmapsProgress} AppMap${
+        this.appmapsProgress !== 1 ? 's' : ''
+      } created`;
+    },
+  },
+
+  methods: {
+    prevStep() {
+      if (this.currentStep > 1) {
+        this.currentStep -= 1;
+      }
+    },
+    nextStep() {
+      if (this.currentStep < this.stepsCount) {
+        this.currentStep += 1;
+      }
+    },
+    async runAction() {
+      console.log(this.onAction);
+      if (typeof this.onAction !== 'function') {
+        return;
+      }
+
+      this.isActionRunning = true;
+
+      try {
+        await this.onAction(
+          this.selectedLanguage,
+          this.currentStep,
+          this.selectedTestFramework
+        );
+      } catch (e) {
+        console.error(e);
+      }
+
+      this.isActionRunning = false;
```

Comment:
I think it makes sense now to wait for an external system to change the state. Can we display the loading indicator regardless of when `this.onAction` returns? It should only disappear once the state has changed.